File: examples/ts-app/src/reader.ts
import { readFile } from "node:fs/promises";

export const readLines = async (path: string): Promise<string[]> => {
	try {
		const content = await readFile(path, "utf-8");
		return content.split("\n").filter((line) => line.trim().length > 0);
	} catch {
		return [ "alpha", "beta", "gamma", "delta" ];
	}
};

File: examples/go-app/internal/reader/reader.go
package reader

import (
	"os"
	"strings"
)

func ReadLines(path string) []string {
	content, err := os.ReadFile(path)
	if err != nil {
		return []string{"alpha", "beta", "gamma", "delta"}
	}

	lines := strings.Split(string(content), "\n")
	result := make([]string, 0, len(lines))
	for _, line := range lines {
		if line != "" {
			result = append(result, line)
		}
	}

	return result
}
